Build fails on -Duse64bitint -DEBUGGING on 32-bit target

**Description**
I've tried to build `blead` (d7c0b58cf68aeb7b08eee56a9e693a161f3f9106) for 32-bit x86 with `-Duse64bitint -DEBUGGING`, but build failed:
```
$ ./Configure -de -Dusedevel -DEBUGGING -Dcc='gcc -m32' -Duse64bitint
$ make
...
./miniperl -w -Ilib -Idist/Exporter/lib -MExporter -e '<?>' || sh -c 'echo >&2 Failed to build miniperl.  Please run make minitest; exit 1'
miniperl: mg.c:3659: S_restore_magic: Assertion `sizeof((IV)(unsigned long)(p)) == sizeof(SSize_t)' failed.
Aborted
Failed to build miniperl. Please run make minitest
make: *** [makefile:377: lib/buildcustomize.pl] Error 1
```
Bisecting points to commit 1a2b24d857d6903bb734c2fcb6cc2a72e1007059.

**Steps to Reproduce**
```
$ ./Configure -de -Dusedevel -DEBUGGING -Dcc='gcc -m32' -Duse64bitint
$ make
```

**Expected behavior**
Build succeeds.

--- a/scope.h
+++ b/scope.h
@@ -283,8 +283,8 @@ casts it to a pointer of that C<type>.
     (I32)(align - ((size_t)((caddr_t)&PL_savestack[PL_savestack_ix]) % align)) % align)
 #define SSNEWat(n,t,align)	SSNEWa((n)*sizeof(t), align)
 
-#define SSPTR(off,type)         (assert(sizeof(off) == sizeof(SSize_t)), (type)  ((char*)PL_savestack + off))
-#define SSPTRt(off,type)        (assert(sizeof(off) == sizeof(SSize_t)), (type*) ((char*)PL_savestack + off))
+#define SSPTR(off,type)         (assert(sizeof(off) >= sizeof(SSize_t)), (type)  ((char*)PL_savestack + off))
+#define SSPTRt(off,type)        (assert(sizeof(off) >= sizeof(SSize_t)), (type*) ((char*)PL_savestack + off))
 
 #define save_freesv(op)		save_pushptr((void *)(op), SAVEt_FREESV)
 #define save_mortalizesv(op)	save_pushptr((void *)(op), SAVEt_MORTALIZESV)

